### PR TITLE
feat(wakunode2): support log format format selection

### LINKED ISSUE #1195: feat: cli flag for selecting log format
### Problem
It would be nice to be able to select log format without having to recompile the binary like `nimbus-eth2` does with `--log-format`:
```nim
    logStdout* {.
      hidden
      desc: "Specifies what kind of logs should be written to stdout (auto, colors, nocolors, json)"
      defaultValueDesc: "auto"
      defaultValue: StdoutLogKind.Auto
      name: "log-format" .}: StdoutLogKind
```
https://github.com/status-im/nimbus-eth2/blob/f7db50522b299ceb558e81342f894ed8f0ac11c7/beacon_chain/conf.nim#L125-L130

### Acceptance criteria
It does what I said.

### PR DESCRIPTION
**Pull request by LNSD**
This PR adds support for structured logging to wakunode2.

- [x] Support configuring 
- [x] Support the [`NO_COLOR` environment variable](HTTP://no-color.org) to disable the colored logs.
- [x] Ammend the current waku logs to support the JSON log format.
- [x] Remove `basic2` example. It has been superseded by the `publish/subscribe` examples.
- [x] Rename `wakubridge` configuration file, and add support for structured logging. 

image

This PR resolves #1195

### CODE REVIEW

**Comment by LNSD** [comment]
This PR depends on https://github.com/status-im/nim-eth/pull/568

**Review by jm-clius — APPROVED** [review]
Good job!

File: apps/wakunode2/config.nim [hunk 1]
```diff
@@ -27,22 +27,24 @@ type ConfResult*[T] = Result[T, string]
...
+      desc: "Sets the log level for process. Supported levels: TRACE, DEBUG, INFO, NOTICE, WARN, ERROR or FATAL",
+      defaultValue: logging.LogLevel.INFO,
+      name: "log-level" .}: logging.LogLevel
 
-    version* {.
-      desc: "prints the version"
-      defaultValue: false
-      name: "version" }: bool
+    logFormat* {.
+      desc: "Specifies what kind of logs should be written to stdout. Suported formats: Text, JSON",
```
[the remark this answers is not in the document]
Comment (LNSD) [reply apps/wakunode2/config.nim:42]: It shouldn't, as per `nim-confutils` documentation.

[the remark this answers is not in the document]
Comment (jakubgs) [reply apps/wakunode2/config.nim:42]: Cool.

[the remark this answers is not in the document]
Comment (LNSD) [reply apps/wakunode2/config.nim:42]: I double-checked, and it seems that `nim-confutils` does not support case insensitivity in `enum` types. And it does not allow to override of the `parseCmdArg` method for other enums.

**Pull request merged**